src/DataValues/TimeValue.php

```diff
@@ -140,9 +140,10 @@ public function __construct( $time, $timezone, $before, $after, $precision, $cal
 		}
 
 		$year = $matches[2];
+		$year = ltrim( $year, '0' );
 		$year = str_pad( $year, 4, '0', STR_PAD_LEFT );
 
-		$this->time = $matches[1] . $year . $matches[3];
+		$this->timestamp = $matches[1] . $year . $matches[3];
```

Comment:
I *really* dislike doing normalization here. The constructor should do basic validation, nothing more. But I see that that's not the topic of this patch.